junit5-support/src/main/kotlin/in/specmatic/test/reports/coverage/OpenApiCoverageReportInput.kt

```diff
@@ -0,0 +1,152 @@
+package `in`.specmatic.test.reports.coverage
+
+import `in`.specmatic.conversions.convertPathParameterStyle
+import `in`.specmatic.core.TestResult
+import `in`.specmatic.core.pattern.ContractException
+import `in`.specmatic.test.API
+import `in`.specmatic.test.TestResultRecord
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+class OpenApiCoverageReportInput(
+    private val testResultRecords: MutableList<TestResultRecord> = mutableListOf(),
+    private val applicationAPIs: MutableList<API> = mutableListOf(),
+    private val excludedAPIs: MutableList<String> = mutableListOf()
+) {
+    fun addTestReportRecords(testResultRecord: TestResultRecord) {
+        testResultRecords.add(testResultRecord)
+    }
+
+    fun addAPIs(apis: List<API>) {
+        applicationAPIs.addAll(apis)
+    }
+
+    fun addExcludedAPIs(apis: List<String>){
+        excludedAPIs.addAll(apis)
+    }
+
+    fun generate(): OpenAPICoverageReport {
+        // If we get a failed test whose path and method are not found in the actuator, it means that the operation exists in the spec, but has not been implemented.
+        val notImplementedTests =
+            testResultRecords.filter { it.result == TestResult.Failed && applicationAPIs.none { api -> api.path == it.path && api.method == it.method } }
+        val testResults = testResultRecords.minus(notImplementedTests.toSet())
+            .plus(notImplementedTests.map { it.copy(result = TestResult.NotImplemented) })
+
+        var allAPITests = addTestResultsForMissingEndpoints(testResults)
+        allAPITests = sortByPathMethodResponseStatus(allAPITests)
+
+        // Creates a structure which looks like this:
```

Comment:
Same as above, can we delete comment and extract method?

Reply:
Fixed.